test/integration/targets/setup_postgresql_db/tasks/main.yml

```diff
@@ -63,9 +63,9 @@
   when: ansible_os_family == "RedHat" and ansible_service_mgr != "systemd"
 
 - name: Initialize postgres (Debian)
-  command: /usr/bin/pg_createcluster {{ pg_ver }} main
-  # Sometimes package install creates the db cluster, sometimes this step is needed
-  ignore_errors: True
+  shell: '. /usr/share/postgresql-common/maintscripts-functions && set_system_locale && /usr/bin/pg_createcluster -u postgres {{ pg_ver }} main'
```

Comment:
You're right, this way isn't reliable, packaging might evolve: instead of mimic postinst script, reusing it is better.

I removed this task and added another one which purges PostgreSQL packages if they are installed.